membership.lua

```diff
@@ -562,25 +574,37 @@ end
 local function init(advertise_host, port)
     checks('string', 'number')
 
-    _sock = socket('AF_INET', 'SOCK_DGRAM', 'udp')
-    local ok = _sock:bind('0.0.0.0', port)
-    if not ok then
-        local err = string.format('Socket bind error: %s', _sock:error())
-        log.error('%s', err)
-        error(err)
+    if _sock == nil or _sock:name().port ~= port then
+        local sock = socket('AF_INET', 'SOCK_DGRAM', 'udp')
+        local ok = sock:bind('0.0.0.0', port)
+        if not ok then
+            local err = string.format('Socket bind error: %s', _sock:error())
+            log.error('%s', err)
```

Comment:
nit: you can avoid usage of format string because your string is already formatted. Just use `log.error(err)`